src/theory/quantifiers/ceg_t_instantiator.cpp

```diff
@@ -1360,6 +1363,198 @@ Node BvInstantiator::rewriteTermForSolvePv(
 }
 
+/** sort bv extract interval
+ *
+ * This sorts lists of bitvector extract terms where
+ * ((_ extract i1 i2) t) < ((_ extract j1 j2) t)
+ * if i1>j1 or i1=j1 and i2>j2.
+ */
+struct SortBvExtractInterval
+{
+  bool operator()(Node i, Node j)
+  {
+    Assert(i.getKind() == BITVECTOR_EXTRACT);
+    Assert(j.getKind() == BITVECTOR_EXTRACT);
+    BitVectorExtract ie = i.getOperator().getConst<BitVectorExtract>();
+    BitVectorExtract je = j.getOperator().getConst<BitVectorExtract>();
+    if (ie.high > je.high)
+    {
+      return true;
+    }
+    else if (ie.high == je.high)
+    {
+      Assert(ie.low != je.low);
+      return ie.low > je.low;
+    }
+    return false;
+  }
+};
+
+void BvInstantiatorPreprocess::registerCounterexampleLemma(
+    std::vector<Node>& lems, std::vector<Node>& ce_vars)
+{
+  // new variables
+  std::vector<Node> vars;
+  // new substitution, will apply { subs -> vars }
+  std::vector<Node> subs;
+  // new lemmas
+  std::vector<Node> new_lems;
+
+  if (options::cbqiBvRmExtract())
+  {
+    Trace("cegqi-bv-pp") << "-----remove extracts..." << std::endl;
+    d_extract_map.clear();
+    std::unordered_set<TNode, TNodeHashFunction> visited;
+    for (unsigned i = 0, size = lems.size(); i < size; i++)
+    {
+      Trace("cegqi-bv-pp-debug2") << "Register ce lemma # " << i << " : "
+                                  << lems[i] << std::endl;
+      process(lems[i], visited);
+    }
+    for (std::pair<const Node, std::vector<Node> >& es : d_extract_map)
+    {
+      if (es.second.size() > 1)
+      {
+        // sort based on the extract start position
+        std::vector<Node> curr_vec;
+        curr_vec.insert(curr_vec.end(), es.second.begin(), es.second.end());
+
+        SortBvExtractInterval sbei;
+        std::sort(curr_vec.begin(), curr_vec.end(), sbei);
+
+        std::map<Node, unsigned> bounds[2];
+        Trace("cegqi-bv-pp") << "For term " << es.first << " : " << std::endl;
+        for (unsigned i = 0, size = curr_vec.size(); i < size; i++)
+        {
+          Trace("cegqi-bv-pp") << "  " << i << " : " << curr_vec[i]
+                               << std::endl;
+          BitVectorExtract e =
+              curr_vec[i].getOperator().getConst<BitVectorExtract>();
+          bounds[0][curr_vec[i]] = e.high;
+          bounds[1][curr_vec[i]] = e.low;
+        }
+
+        bool success = false;
+        TypeNode tn = es.first.getType();
+        Assert(tn.isBitVector());
+        unsigned width = tn.getBitVectorSize();
+        Trace("cegqi-bv-pp") << "Width is " << width << std::endl;
+        do
+        {
+          success = false;
+          std::vector<Node> next_vec;
+          std::vector<Node> tmp_vars;
+          std::vector<Node> tmp_subs;
+          // the children of the concat we will construct
+          std::vector<Node> children;
+          unsigned curr_index = width;
+          for (unsigned i = 0, size = curr_vec.size(); i < size; i++)
+          {
+            Node n = curr_vec[i];
```

Comment:
You don't need `n` here, just use index `i` to access `bounds_high[i]` and `bounds_low[i]`.